java/vector/src/main/java/org/apache/arrow/vector/ValueVector.java

```diff
@@ -237,4 +237,10 @@
    * @return true if element is null
    */
   boolean isNull(int index);
+
+  //TODO remove default and implement in subclasses
+  int hashCode(int index);
+
+  //TODO remove default and implement in subclasses
```

Comment:
it looks like this and the above TODO are done?  Or are more necessary?

Reply:
Thanks for your reminder, removed.